src/app/api/faucet/verify/route.ts

```diff
@@ -0,0 +1,60 @@
+import { NextResponse } from 'next/server';
+import { initializeClients, publicClient } from '../../../../lib/client';
+import { abi, address } from '../../../../lib/constants';
+
+async function verify(userAddress: string) {
+    try {
+        initializeClients();
+
+        if (!publicClient) {
+            throw new Error('Public client is not initialized');
+        }
+
+        type CanWithdrawResult = [boolean, string, bigint];
+        type WithdrawalDelayResult = bigint;
+
+        const [canWithdrawResult, reason, lastWithdrawalTime] = await publicClient.readContract({
+            address,
+            abi,
+            functionName: 'canWithdraw',
+            args: [userAddress],
+        }) as CanWithdrawResult;
+
+        const withdrawalDelay = await publicClient.readContract({
+            address,
+            abi,
+            functionName: 'withdrawalDelay',
+            args: [],
+        }) as WithdrawalDelayResult;
+
+        const formattedLastWithdrawalTime = lastWithdrawalTime.toString();
+        const formattedWithdrawalDelay = withdrawalDelay.toString();
+
+        if (canWithdrawResult) {
```

Comment:
there is unnecessary nesting here

```
if (condition) {

   return 
} else {

   return
}
```


is the same as 


```
if (condition) {

return
}

return

```

The else is redundant if the if statement has a return in it